### PR TITLE
Perpetuals recent trades table

### PR DESCRIPTION
**Pull request by tiltom**

### CODE REVIEW

**Comment by creed-victor** [comment]
This supposed to be perpetual swaps or margin trading?

**Comment by ShynRou** [comment]
@creed-victor It's a sub-feature for the new perpetuals page.

File: src/app/pages/PerpetualPage/components/RecentTradesTable/components/RecentTablesRow/index.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          : 'tw-text-trade-long',
+        isOddRow && 'tw-bg-gray-3',
+        'tw-h-6',
+      )}
+    >
+      <td
+        className={classNames(
+          'tw-px-4 tw-py-1 tw-text-right tw-font-semibold tw-flex',
+          isOddRow && 'tw-rounded-l',
+          priceColor && priceColor,
```
Comment (ShynRou) [inline src/app/pages/PerpetualPage/components/RecentTradesTable/components/RecentTablesRow/index.tsx:37]: && is unnecessary here.

File: src/app/pages/PerpetualPage/components/RecentTradesTable/index.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  pair: PerpetualPair;
+};
+
+export const RecentTradesTable: React.FC<IRecentTradesTableProps> = ({
+  pair,
+}) => {
+  const data = usePerpetual_RecentTradesTable();
+
+  return (
+    <table className={styles['recent-trades-table']}>
```
Comment (ShynRou) [inline src/app/pages/PerpetualPage/components/RecentTradesTable/index.tsx:20]: use camelCase
styles.recentTradesTable

File: src/app/pages/PerpetualPage/hooks/usePerpetual_RecentTradesTable.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  return new Promise(resolve => resolve(result));
+};
+
+export const usePerpetual_RecentTradesTable = () => {
+  const blockId = useBlockSync();
+  const [data, setData] = useState<RecentTradesDataEntry[] | null>();
+
+  useEffect(() => {
+    generatePlaceholderData().then(data => {
```
Comment (ShynRou) [inline src/app/pages/PerpetualPage/hooks/usePerpetual_RecentTradesTable.ts:51]: Please add a TODO comment, when adding placeholder functions.

**Review by ShynRou — CHANGES_REQUESTED** [review]

File: src/app/pages/PerpetualPage/components/RecentTradesTable/index.tsx (hunk 2 above)
Comment (tiltom) [reply src/app/pages/PerpetualPage/components/RecentTradesTable/index.tsx:20]: I will in this case, but don't we want to use BEM when we are writing styles?

Comment (ShynRou) [reply src/app/pages/PerpetualPage/components/RecentTradesTable/index.tsx]: BEM is not needed with Module SCSS, since classes are hashed anyway.

**Review by ShynRou — APPROVED** [review]

**Pull request merged**